operator/src/main/java/org/keycloak/operator/controllers/WatchedSecretsStore.java

```diff
@@ -111,13 +111,12 @@ public void addLabelsToWatchedSecrets() {
 
                 Log.infof("Adding label to Secret \"%s\"", secret.getMetadata().getName());
 
-                secret = new SecretBuilder(secret)
-                        .editMetadata()
-                        .addToLabels(Constants.KEYCLOAK_COMPONENT_LABEL, WATCHED_SECRETS_LABEL_VALUE)
-                        .endMetadata()
-                        .build();
-
-                client.secrets().inNamespace(secret.getMetadata().getNamespace()).withName(secret.getMetadata().getName()).patch(secret);
+                client.secrets().inNamespace(secret.getMetadata().getNamespace()).withName(secret.getMetadata().getName())
```

Comment:
I'm afraid the `secret.getMetadata().getNamespace()` can throw NPE. As the code is executed even when the metadata are null for the secret.

Reply:
Yep, you're right. I've been thinking about it in a more fault-tolerant way as after some invalid secret creation in a cluster, the whole process could be canceled. I was thinking about some warning dump and ignoring the secret, but it's probably better not to continue and solve the issue with that. Moreover, without a name, the secret wouldn't even be fetched, so the metadata are always there.

IMHO, the PR is ready to go. Thanks!